pkg/opensearch/response_parser.go

```diff
@@ -1360,34 +1370,39 @@ func getErrorFromOpenSearchResponse(response *client.SearchResponse) error {
 	return err
 }
 
-func getKeyTraceSpanValuePairs(source map[string]interface{}) []map[string]interface{} {
-	transformedAttributes := []map[string]interface{}{}
+func getTraceKeyValuePairs(source map[string]interface{}) []KeyValue {
+	transformedAttributes := []KeyValue{}
 	for k, v := range source {
-		transformedAttributes = append(transformedAttributes, map[string]interface{}{"key": k, "value": v})
+		transformedAttributes = append(transformedAttributes, KeyValue{Key: k, Value: v})
 	}
 	return transformedAttributes
 }
 
-func transformTraceEventsToLogs(events []interface{}) ([]map[string]interface{}, []string, error) {
-	spanEvents := []map[string]interface{}{}
+func transformTraceEventsToLogs(events []interface{}) ([]Log, []string, error) {
+	spanEvents := []Log{}
 	stackTraces := []string{}
 	if len(events) > 0 {
 		for _, event := range events {
-			eventObj := event.(map[string]interface{})
-			timeStamp, err := utils.TimeFieldToMilliseconds(eventObj["time"])
-			if err != nil {
-				return nil, nil, err
-			}
-			spanEvents = append(spanEvents, map[string]interface{}{"timestamp": timeStamp, "fields": []map[string]interface{}{{"key": "name", "value": eventObj["name"]}}})
+			eventFields := []KeyValue{}
+			if eventObj, exists := event.(map[string]interface{}); exists {
+				if eventAttributes, exists := eventObj["attributes"].(map[string]interface{}); exists {
+					eventFields = getTraceKeyValuePairs(eventAttributes)
+				}
+				timeStamp, err := utils.TimeFieldToMilliseconds(eventObj["time"])
+				if err != nil {
+					return nil, nil, err
+				}
+				spanEvents = append(spanEvents, Log{Timestamp: timeStamp, Name: eventObj["name"].(string), Fields: eventFields})
 
-			// get stack traces if error event
-			attributes, ok := eventObj["attributes"].(map[string]interface{})
-			if ok {
-				errorValue := attributes["error"]
-				if errorValue != nil {
-					stackTraces = append(stackTraces, fmt.Sprintf("%s: %s", eventObj["name"], attributes["error"]))
+				// get stack traces if error event
+				if attributes, exists := eventObj["attributes"].(map[string]interface{}); exists {
```

Comment:
nit: could this be combined with the for loop on line 1387? or do we think its more readable this way

Reply:
Done!